tasks/systemd.yml

```diff
@@ -7,19 +7,11 @@
     owner: root
     group: root
     mode: 0755
-  register: systemd_service_file
   notify: restart haproxy
 
 - name: Systemd - Ensure service is linked into /etc/systemd.
   file:
     src: "/lib/systemd/system/haproxy.service"
     dest: "/etc/systemd/system/multi-user.target.wants/haproxy.service"
     state: link
-
-#
-# Keep systemd-specific bits here instead of pushing a global handler.
-#
-- name: Systemd - Ensure systemd daemon is reloaded.
-  systemd:
-    daemon_reload: yes
-  when: systemd_service_file.changed
+  notify: reload systemd
```

Comment:
This notify should be before the `notify: restart haproxy`.

Reply:
No, handlers are executed in the order of declaration in `notify/main.yml` *not* in the order they are called.